stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/AddressElement.kt

```diff
@@ -60,7 +60,7 @@ open class AddressElement(
         PhoneNumberController.createPhoneNumberController(
             initialValue = rawValuesMap[IdentifierSpec.Phone] ?: "",
             showOptionalLabel = addressType.phoneNumberState == PhoneNumberState.OPTIONAL,
-            acceptAnyInput = true,
+            acceptAnyInput = addressType.phoneNumberState != PhoneNumberState.REQUIRED,
```

Comment:
This won't affect `PaymentSheet` as it has separate fields for collecting phone numbers and is only required if `collectPhone` is true in `BillingDetailsCollectionConfiguration`. Collecting full billing details does not expose the phone number field as it does in the `AddressElement` product.